lib/schematype.js

```diff
@@ -139,6 +150,10 @@ SchemaType.cast = function cast(caster) {
   return this._cast;
 };
 
+SchemaType.set = function set(option, value) {
+  this.defaultOptions[option] = value;
```

Comment:
Let's add a check here for `this.hasOwnProperty('defaultOptions')` to ensure that `set()` never modifies the base class's default options. Something like:

```
if (!this.hasOwnProperty('defaultOptions')) {
  this.defaultOptions = Object.assign({}, this.defaultOptions);
}
this.defaultOptions[option] = value;
```

I ran into this issue a lot when implementing support for `SchemaNumber.cast()`, etc. See [Static Properties in JavaScript Classes with Inheritance](http://thecodebarbarian.com/static-properties-in-javascript-with-inheritance.html) for some learnings on that. Also see [`hasOwnProperty()`](https://masteringjs.io/tutorials/fundamentals/hasownproperty).